### ISSUE TITLE
Azure database exporter not working

### ISSUE THREAD

**Issue by ctradar** [percona/pmm#3095]
### Description

Azure database exporter not working. Follow [this](https://docs.percona.com/percona-monitoring-and-management/setting-up/client/azure.html
[AZURE_DATABASE_EXPORTER f90f739c-c0de-4966-b626-7c0dadf4ab35.log](https://github.com/user-attachments/files/16371219/AZURE_DATABASE_EXPORTER.f90f739c-c0de-4966-b626-7c0dadf4ab35.log)
) guide

### Expected Results

Fetch metrics from Azure successfully

### Actual Results

Failing with error 4xx

### Version

2.38.0 (tried with version 2.30.0 and latest version 2.42.0 as well)

### Steps to reproduce

Enable Azure Monitoring technical preview feature. Add managed database service and enable Azure metrics checkbox while doing so.

### Relevant logs

```Shell
ERRO[2024-07-25T05:47:11.939+00:00] time="2024-07-25T05:47:11Z" level=error msg="Received 400 status for resource /subscriptions/redacted/resourceGroups/redacted/providers/Microsoft.DBforPostgreSQL/flexibleServers/redacted3/providers/microsoft.insights/metrics?aggregation=Average&api-version=2018-01-01&metricnames=cpu_percent%2Cmemory_percent%2Cstorage_percent%2Cstorage_used%2Cstorage_limit%2Cnetwork_bytes_egress%2Cnetwork_bytes_ingress&timespan=2024-07-25T05%3A43%3A11Z%2F2024-07-25T05%3A44%3A11Z. " source="main.go:58"  agentID=/agent_id/f90f739c-c0de-4966-b626-7c0dadf4ab35 component=agent-process type=azure_database_exporter
```

### Code of Conduct

- [X] I agree to follow Percona Community Code of Conduct

**Comment by ctradar** [comment]
[AZURE_DATABASE_EXPORTER f90f739c-c0de-4966-b626-7c0dadf4ab35.log](https://github.com/user-attachments/files/16371285/AZURE_DATABASE_EXPORTER.f90f739c-c0de-4966-b626-7c0dadf4ab35.log)

Logs

**Comment by Nailya** [comment]
@ctradar i couldn't reproduce this issue. Could you check if the database is running and all credentials are correct?

**Comment by nkprince007** [comment]
Yes, the database was indeed running and using correct azure service principal credentials.
